src/libraries/System.Collections/src/System/Collections/Generic/PriorityQueue.cs

```diff
@@ -961,9 +961,11 @@ void IEnumerator.Reset()
             /// <returns>An <see cref="Enumerator"/> for the <see cref="UnorderedItems"/>.</returns>
             public Enumerator GetEnumerator() => new Enumerator(_queue);
 
-            IEnumerator<(TElement Element, TPriority Priority)> IEnumerable<(TElement Element, TPriority Priority)>.GetEnumerator() => GetEnumerator();
+            IEnumerator<(TElement Element, TPriority Priority)> IEnumerable<(TElement Element, TPriority Priority)>.GetEnumerator() =>
+                _queue.Count == 0 ? ((IEnumerable<(TElement Element, TPriority Priority)>)Array.Empty<(TElement Element, TPriority Priority)>()).GetEnumerator() :
```

Comment:
Nit: it might not be immediately evident to the reader without context that this expression is intended to return a singleton enumerator. Perhaps extract to a shared helper method named appropriately?
```suggestion
                _queue.Count == 0 ? GetEmptyEnumeratorSingleton<(TElement Element, TPriority Priority)>() :
```

Reply:
Added